### PR TITLE
[FEATURE] Adding support for PostgreSQL

### PR DESCRIPTION
**Pull request by M4ST3RX**
#### Issue or feature explanation
This feature adds support for PostgreSQL database connections. See #46 

#### Proposed solution/change
See commits for the changes. Few important bits:

- Created Prequel facade as a universal query builder
- Changed usage of DB to custom PDB facade
- Added an argument to `DatabaseConnector::getConnection` so PostgreSQL can query multiple databases.
- Added `getTableData` function to retrieve the rows of a table.

### CODE REVIEW

File: src/Classes/App/AppStatus.php [hunk 1]
```diff
@@ -73,7 +85,12 @@ private function serverInfo(): array
      */
     public function userPermissions(): array
     {
-        $grants      = (array)$this->connection->select('SHOW GRANTS FOR CURRENT_USER();')[0];
+        //PostgreSQL
+        if($this->connection->getPdo()->getAttribute(PDO::ATTR_DRIVER_NAME) === 'pgsql') {
```
Comment (QuintenJustus) [inline src/Classes/App/AppStatus.php]: Better to be consistent when checking database driver, use `config('prequel.database.connection') === "pgsql"` like the other ones.

File: src/Classes/App/AppStatus.php [hunk 2]
```diff
@@ -54,16 +54,28 @@ public function getStatus(): array
...
+            $serverInfoArray    = [];
+
+            foreach ($explodedServerInfo as $attr) {
+                $split                 = explode(': ', $attr);
+                $key                   = strtoupper(str_replace(' ', '_', str_replace(':', '', $split[0])));
+                $serverInfoArray[$key] = $split[1];
+            }
         }
+        //$serverInfo = $this->connection->getPdo()->query('select extract(epoch from current_timestamp - pg_postmaster_start_time())')->fetch()[0];
+//        echo "<pre>".print_r($serverInfo, true)."</pre>";
```
Comment (QuintenJustus) [inline src/Classes/App/AppStatus.php]: Tip: you can use the [dd(...args)](https://laravel.com/docs/5.8/helpers#method-dd) function to debug your data!

Comment (M4ST3RX) [reply src/Classes/App/AppStatus.php]: i use a custom helper called `pr` which does that echo thing but I put it into the project later and that stayed there for no reason. Will take it out in a second.

Comment (M4ST3RX) [reply src/Classes/App/AppStatus.php]: `//pr($serverInfoArray);` this is just under it because by then I imported my helpers.

**Comment by QuintenJustus** [comment]
Looks good!

**Comment by James-N-M** [comment]
@M4ST3RX next time you take on a issue like this lets pair it, I've love to learn how you went about it 💯

**Pull request merged**